### PR TITLE
Merged yolo_detector and object detection sample

### PR DESCRIPTION
**Pull request by gursimarsingh**
Relates to #25006 

This pull request merges the yolo_detector.cpp sample with the object_detector.cpp sample. It also beautifies the bounding box display on the output images

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
There is some rebase/merge issue. I see a lot of irrelevant changes.

**Comment by gursimarsingh** [comment]
@asmorkalov I see only 7 files being changed and I have pulled latest 5.x to this branch

**Comment by vpisarev** [comment]
@gursimarsingh, thank you!
the sample works correctly, at least on some variations of Yolo.

Here are my comments:

1. I don't see any output on static images from opencv/samples/data.
2. What's the method to get list of detection models?
3. Visualization of boxes is bad: colors are not bright, label background does not have the same color as boxes, lines are too thin and fonts are too small on high-resolution images
4. To initialize font inside callback for each frame and each box is very bad idea. Please, initialize the font once outside of the loop and pass reference to it.
5. `postprocess() combines post-processing itself, as well as visualization. We need clear function to produce `std::vector<Rect>& boxes, std::vector<std::string>& labels, std::vector<float>& conf`. And then a separate function to draw those boxes.

**Comment by gursimarsingh** [comment]
@vpisarev 
1. --async in the current implementation is ignored and instead HAVE_THREADS is used to control the asynchronous behavior. Will make the change to use this input argument as a control between async and sync mode
2.  On static images the code is running. Yolov8n does not work on some images but that is related to model quality. 
`./bin/example_dnn_object_detection yolov8n --input=../samples/data/messi5.jpg ` works and looks like this:
image
3. findAliasis function has been added in common.hpp and printAliasis has been added in object_detection.cpp
4. Visualization has been updated. Sample:
frame
5. Font initialization has been fixed
6. postprocess() and drawPred() functions have been separated

Other changes: 
1. Duplicate yolov8 has been removed from models.yaml and replaced with yolov8n.

**Comment by gursimarsingh** [comment]
Bounding box text have been updated along with FPS text

image (7)
image (8)

**Review by vpisarev — APPROVED** [review]

**Comment by vpisarev** [comment]
@asmorkalov, the demo is in great shape, can be merged!

**Review by vpisarev — APPROVED** [review]

File: samples/dnn/object_detection.cpp [hunk 1]
```diff
@@ -405,35 +573,67 @@ void postprocess(Mat& frame, const std::vector<Mat>& outs, Net& net, int backend
...
+    else if (outLayerType == "Identity")
+    {
+        //![forward_buffers]
+        vector<int> keep_classIds;
+        vector<float> keep_confidences;
+        vector<Rect2d> keep_boxes;
+        //![forward_buffers]
+
+        //![postprocess]
+        yoloPostProcessing(outs, keep_classIds, keep_confidences, keep_boxes, confThreshold, nmsThreshold, "yolov8");
```
[the remark this answers is not in the document]
Comment (asmorkalov) [reply samples/dnn/object_detection.cpp:585]: The same relevant for python code. Also there conditions are different.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**